src/plugins/defoldcollection/defoldcollectionplugin.cpp

```diff
@@ -0,0 +1,473 @@
+            // the value will be automatically correct.
+            map_h["tile_set"] = "/tilesources/" + tileset_current->name() + ".tilesource";
+
+            QString result = replaceTags(QLatin1String(map_t), map_h);
+            Tiled::SaveFile mapFile(tilesetFilePath);
+            if (!mapFile.open(QIODevice::WriteOnly | QIODevice::Text))
+            {
+                mError = tr("Could not open file for writing.");
+                return false;
+            }
+            QTextStream stream(mapFile.device());
+            stream << result;
+
+            if (mapFile.error() != QFileDevice::NoError)
+            {
+                mError = mapFile.errorString();
+                return false;
+            }
+
+            if (!mapFile.commit())
+            {
+                mError = mapFile.errorString();
+                return false;
+            }
+        }
+
+        QVariantHash collection_h;
+        collection_h["children"] = QString("");
+        collection_h["embedded-instances"] = QString("");
+        // optional custom properties that allow exporting a map with an offset
+        collection_h["pos-x"] = map->property(QLatin1String("x-offset")).toInt();
+        collection_h["pos-y"] = map->property(QLatin1String("y-offset")).toInt();
+        collection_h["components"] = components;        
+
+        QString result = replaceTags(QLatin1String(collection_t), collection_h);
+        Tiled::SaveFile mapFile(fileName);
+        if (!mapFile.open(QIODevice::WriteOnly | QIODevice::Text))
+        {
+            mError = tr("Could not open file for writing.");
+            return false;
+        }
+        QTextStream stream(mapFile.device());
+        stream << result;
+
+        if (mapFile.error() != QFileDevice::NoError)
+        {
+            mError = mapFile.errorString();
+            return false;
+        }
+
+        if (!mapFile.commit()) {
+            mError = mapFile.errorString();
+            return false;
+        }
+    }
+    // if there are group layers, each group layer is a gameobject, parented to the top gameobject, while being a parent to multiple tilemap components
+    else
+    {
+        QString children;
+        QString embeddedInstances;
+
+        // foreach Group Layer
+        int groupLayerOrder = 0;
+        for (auto& layer : map->layers())
+        {
+            if (layer->layerType() != Tiled::Layer::GroupLayerType)
+                continue;
+
+            QVariantHash child_h;
+            child_h["child-name"] = layer->name();
+            children.append(replaceTags(QLatin1String(child_t), child_h));
+
+            QVariantHash emdedded_instance_h;
+            emdedded_instance_h["instance-name"] = layer->name();
+
+            QString components;
+
+            // write as many tilemaps as there are tilesets per group layer
+            for (auto& tileset_current : map->tilesets())
+            {
+                QString tilesetFilePath = tilesetFileDir;
```

Comment:
Even though you're writing one tilemap per tileset, this is still a tile map file, so please rename to `tilemapFilePath` to avoid confusion.